src/shell/mod.rs

```diff
@@ -435,6 +429,12 @@ impl WorkspaceSet {
             }
         }
     }
+
+    pub fn update_amount(&mut self, amount: WorkspaceAmount) {
+        self.amount = amount;
+        // TODO
+        // - what if there are more workspace than amount?
```

Comment:
Merge the last couple of workspaces. There should be code somewhere in this file, that does this (perhaps for the mode change? not sure).